src/sipsess/sess.c

```diff
@@ -372,6 +372,21 @@ bool sipsess_ack_pending(const struct sipsess *sess)
 }
 
 
+/**
+ * Compares the sessions SIP message with given SIP message
+ *
+ * @param sess  SIP Session
+ * @param msg   SIP Message
+ * @return True if msg matches with sess->msg
+ */
+bool sipsess_msg_equal(const struct sipsess *sess, const struct sip_msg *msg)
+{
+	if (!sess)
+		return false;
+
+	return sess->msg == msg;
```

Comment:
is it more robust to compare the Call-ID instead of a raw pointer ?

Reply:
Maybe yes. I have to check how the re-send mechanism is handled in re/baresip and how the postponed call to ua_accept() may influence this.

Reply:
The `sip->ht_strans` hash table is used to avoid that two SIP messages with equal via.branch are processed twice. This avoids also that two SIP message (with different pointer) with equal call-id are processed twice. This is handled all by `re`.

So I think that comparing the pointers are perfect in order to **avoid** that multiple call objects are allocated for one SIP INVITE.

Reply:
> edit: Comparing call-id would also work. Does it have any benefit?

Let's say you duplicate the `sip_msg` for some reason (multi threading...):

```c
struct sip_msg msg = *msg_orig;
sipsess_msg_equal(csess, &msg);
```

This would fail.